server/util/src/api_helper.rs

```diff
@@ -0,0 +1,42 @@
+use std::time::Duration;
+
+use reqwest::*;
+
+pub struct RetrySeconds(Vec<u64>);
+
+impl Default for RetrySeconds {
+    fn default() -> Self {
+        Self(vec![
+            /* first retry */ 2, /* second retry */ 5, /* third retry */ 10,
+        ])
+    }
+}
+
+pub async fn with_retries<F>(connection_timeouts: RetrySeconds, f: F) -> Result<Response>
+where
+    F: Fn(Client) -> RequestBuilder,
+{
+    let mut index = 0;
+    loop {
+        let client = Client::builder()
+            .connect_timeout(Duration::from_secs(connection_timeouts.0[index]))
+            .build()
+            .unwrap();
+
+        let result = f(client).send().await;
+
+        let (status, is_connect_error) = match result.as_ref() {
+            Ok(r) => (Some(r.status().clone()), false),
+            Err(e) => (e.status().clone(), e.is_connect()),
+        };
+
+        if (is_connect_error || status == Some(StatusCode::REQUEST_TIMEOUT))
+            && (index + 1) < connection_timeouts.0.len()
+        {
```

Comment:
```suggestion
        {
            log::info!(
                "Connection timeout, retrying in {} seconds",
                connection_timeouts.0[index]
            );
```

Not really needed but like the idea of logging things like this that might help support figure out an issue.
What do you think about logging this or similar?

For this you'd need to include `log` in the `Cargo.toml` too

Reply:
> What do you think about logging this or similar?

I think it's good to have logs, but I think this log will just be noise, I can't see how we can action on it? If there is actually a problem connection user should see this in UI (and pass along to support) ?